### ISSUE TITLE
Rename and Coerce do not work together

### ISSUE THREAD

**Issue by aaron11496** [pyeve/cerberus#607]
```python
>>> import cerberus
>>> cerberus.__version__
'1.3.4'
>>> from cerberus.validator import Validator
>>> Validator().normalized({'f': '123'}, schema={'f': {'type': 'integer', 'coerce': int}})
{'f': 123}
>>> Validator().normalized({'f': '123'}, schema={'f': {'type': 'integer', 'coerce': int, 'rename': 'g'}})
{'g': '123'}
```

I would expect the coerce to still take effect here. Please let me know whether this is intended behavior or if there is another way to do what I intended.

**Comment by funkyfuture** [comment]
yes, it is a known issue that people walk into by intuition. be aware that validation rules are validated against the normalized document.

i keep this issue open as a reminder that admonitions that point people to the proper considerations should be added to the docs about schemas and normalization rules. also, the former should be renamed to just "Schemas". other documents and docstrings also still state "validation schemas" (sometimes joined w/ a dash) though they are meant generally.

**Comment by aeoluszzff** [comment]
also rename and default don't work together
i think rename should process at last, 
`    def __normalize_mapping(self, mapping, schema):
        if isinstance(schema, _str_type):
            schema = self._resolve_schema(schema)
        schema = schema.copy()
        for field in schema:
            schema[field] = self._resolve_rules_set(schema[field])
        #FIX: rename first cause default and coerce don't work with pass in fields 
        #self.__normalize_rename_fields(mapping, schema)
        if self.purge_unknown and not self.allow_unknown:
            self._normalize_purge_unknown(mapping, schema)
        if self.purge_readonly:
            self.__normalize_purge_readonly(mapping, schema)
        # Check `readonly` fields before applying default values because
        # a field's schema definition might contain both `readonly` and
        # `default`.
        self.__validate_readonly_fields(mapping, schema)
        self.__normalize_default_fields(mapping, schema)
        self._normalize_coerce(mapping, schema)
        self.__normalize_containers(mapping, schema)
        ######FIX rename last
        self.__normalize_rename_fields(mapping, schema)
        self._is_normalized = True
        return mapping`
